packages/expo-image/ios/EXImageView.m

```diff
@@ -17,9 +36,67 @@ - (void)dealloc
 
 - (void)setSource:(NSDictionary *)source
 {
-  NSURL *imageURL = [RCTConvert NSURL:source[sourceUriKey]];
+  _source = source;
+  _needsReload = YES;
+}
 
-  [self sd_setImageWithURL:imageURL];
+- (void)setResizeMode:(RCTResizeMode)resizeMode
+{
+  if (_resizeMode == resizeMode) return;
+  
+  // Image needs to be reloaded whenever repeat is enabled or disabled
+  _needsReload = _needsReload || (resizeMode == RCTResizeModeRepeat) || (_resizeMode == RCTResizeModeRepeat);
+  _resizeMode = resizeMode;
+  
+  // Repeat resize mode is handled by the UIImage. Use scale to fill
+  // so the repeated image fills the UIImageView.
+  self.contentMode = resizeMode == RCTResizeModeRepeat
+    ? UIViewContentModeScaleToFill
+    : (UIViewContentMode)resizeMode;
+}
+
+- (void)didSetProps:(NSArray<NSString *> *)changedProps;
+{
+  if (_needsReload) {
+    [self reloadImage];
+  }
+}
+
+- (void)reloadImage
+{
+  _needsReload = NO;
+  
+  NSURL *imageUrl = _source ? [RCTConvert NSURL:_source[sourceUriKey]] : nil;
```

Comment:
At first I also used `imageUrl`, but then I noticed that both `NSURL` treats `URL` as full uppercase and there is a `sd_imageURL`, so I figured that must be the Obj-C way of naming URLs (💅)

Reply:
Ah okay. Well this PR focused on the resizeMode part, but I had to make some changes to the loading as well to make that work. We should probably use your loading code from the events PR, and then I can integrate the resizeMode changes in there.